packages/sanity/src/core/releases/tool/components/ReleaseMenuButton/ReleaseMenuButton.tsx

```diff
@@ -19,35 +20,117 @@ export type ReleaseMenuButtonProps = {
 const ARCHIVABLE_STATES = ['active', 'published']
 
 export const ReleaseMenuButton = ({disabled, release}: ReleaseMenuButtonProps) => {
-  const {archive, unarchive} = useReleaseOperations()
+  const toast = useToast()
+  const {archive} = useReleaseOperations()
+  const {loading: isLoadingReleaseDocuments, results: releaseDocuments} = useBundleDocuments(
+    getBundleIdFromReleaseDocumentId(release._id),
+  )
   const [isPerformingOperation, setIsPerformingOperation] = useState(false)
   const [selectedAction, setSelectedAction] = useState<'edit' | 'confirm-archive'>()
 
-  const releaseMenuDisabled = !release || disabled
+  const releaseMenuDisabled = !release || isLoadingReleaseDocuments || disabled
   const {t} = useTranslation(releasesLocaleNamespace)
   const telemetry = useTelemetry()
 
-  const handleArchive = async (e: Parameters<FormEventHandler<HTMLFormElement>>[0]) => {
+  const handleArchive = useCallback(async () => {
     if (releaseMenuDisabled) return
-    e.preventDefault()
 
-    setIsPerformingOperation(true)
-    await archive(release._id)
+    try {
+      setIsPerformingOperation(true)
+      await archive(release._id)
 
-    // it's in the process of becoming true, so the event we want to track is archive
-    telemetry.log(ArchivedRelease)
-    setIsPerformingOperation(false)
-  }
+      // it's in the process of becoming true, so the event we want to track is archive
+      telemetry.log(ArchivedRelease)
+      toast.push({
+        closable: true,
+        status: 'success',
+        title: (
+          <Text muted size={1}>
+            <Translate
+              t={t}
+              i18nKey="toast.archive.success"
+              values={{title: release.metadata.title}}
+            />
+          </Text>
+        ),
+      })
+    } catch (archivingError) {
+      toast.push({
+        status: 'error',
+        title: (
+          <Text muted size={1}>
+            <Translate
+              t={t}
+              i18nKey="toast.archive.error"
+              values={{title: release.metadata.title, error: archivingError.toString()}}
+            />
+          </Text>
+        ),
+      })
+      console.error(archivingError)
```

Comment:
Rogue console log or meant to be here? :)

Reply:
I followed similar approach to what we do in Publish and schedule, so this is intended